.github/workflows/checks.yml

```diff
@@ -3,27 +3,282 @@ name: Checks
 on: [push, pull_request]
 
 jobs:
-  test:
+  linux_tests:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Configure environment variables
+        run: |
+          # echo "::set-env name=SHELLFLAGS::-x"
+          echo "::set-env name=BUILDFLAGS::"
+          echo "::set-env name=RSPY_TARGET_DIR::${GITHUB_WORKSPACE}/target"
+          echo "::set-env name=CARGO_TARGET_DIR::${GITHUB_WORKSPACE}/target"
+      # Necessary for now for the cargo cache:
+      # https://github.com/actions/cache/issues/133#issuecomment-599102035
+      - name: Fix ~/.cache permissions
+        run: |
+          sudo chown -R $(whoami):$(id -ng) ~/.cargo/
+      - name: Cache cargo registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo index
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/git
+          key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo target
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/target
+          key: ${{ runner.os }}-cargo-target-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo rslib
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/rslib/target
+          key: ${{ runner.os }}-cargo-rslib-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache cargo rspy
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/rspy/target
+          key: ${{ runner.os }}-cargo-rspy-${{ hashFiles('**/Cargo.toml') }}-5-
+      - name: Cache pip wheels
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}-5-
+      - name: Cache pyenv
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/pyenv
+          key: ${{ runner.os }}-pyenv-${{ hashFiles('**/Makefile') }}-${{ hashFiles('**/requirements.txt') }}-5-
       - name: Set up python
         uses: actions/setup-python@v1
         with:
           python-version: 3.7
       - name: Set up protoc
-        uses: ankitects/setup-protoc@master
+        uses: arduino/setup-protoc@v1.1.0
```

Comment:
The reason why I forked this repo is because I don't know enough about GitHub's actions architecture to know if it's safe to be exposing the GitHub token to third party actions.

Reply:
I asked them about it on https://github.com/actions/virtual-environments/issues/602

While we do not have an answer, I am going to revert it back to use your fork.